src/linker/Linker/Annotations.cs

```diff
@@ -239,9 +254,14 @@ public bool IsRelevantToVariantCasting (TypeDefinition type)
 			return types_relevant_to_variant_casting.Contains (type);
 		}
 
-		public void Processed (IMetadataTokenProvider provider)
+		public bool SetProcessed (IMetadataTokenProvider provider)
 		{
-			processed.Add (provider);
+			if (processed.Add (provider)) {
```

Comment:
I thought of adding something like MarkProcessed, but it would only be used in one place because MarkStep often does a small amount of extra work for items which are already processed and get marked again.

Reply:
I think I counted more places than 1 and there is also the second combination doing a similar thing with Matk+CheckProcessed. I think it's worth looking if this can be written more effectively

Reply:
You're right - I was looking at CheckProcessed, but there are more for SetProcessed. I will add a helper for these cases.

Reply:
I added a helper for the Mark+SetProcessed case, but I left CheckProcessed alone since derived classes might override it.